agent/src/plugin_manager/v1beta1.rs

```diff
@@ -317,6 +370,27 @@ pub mod device_plugin_client {
                 .server_streaming(request.into_request(), path, codec)
                 .await
         }
+        #[doc = " GetPreferredAllocation returns a preferred set of devices to allocate"]
+        #[doc = " from a list of available ones. The resulting preferred allocation is not"]
+        #[doc = " guaranteed to be the allocation ultimately performed by the"]
+        #[doc = " devicemanager. It is only designed to help the devicemanager make a more"]
+        #[doc = " informed allocation decision when possible."]
+        pub async fn get_preferred_allocation(
+            &mut self,
+            request: impl tonic::IntoRequest<super::PreferredAllocationRequest>,
+        ) -> Result<tonic::Response<super::PreferredAllocationResponse>, tonic::Status> {
+            self.inner.ready().await.map_err(|e| {
```

Comment:
can we use this preferredallocation concept to avoid kubelet requesting shared resources?  (or maybe this isn't worth investing in if we move to DRA)

Reply:
Well the preferred allocation is preference, there are no guarantees that the kubelet will follow those preferences (especially if multiple resources are requested), so I think it's not worth it (esp. compared to DRA)